lib/phoenix_live_view/js.ex

```diff
@@ -218,10 +218,120 @@ defmodule Phoenix.LiveView.JS do
 
   defimpl Phoenix.HTML.Safe, for: Phoenix.LiveView.JS do
     def to_iodata(%Phoenix.LiveView.JS{} = js) do
-      Phoenix.HTML.Engine.html_escape(Phoenix.json_library().encode!(js.ops))
+      js
+      |> JS.to_encodable()
+      |> Phoenix.json_library().encode!()
+      |> Phoenix.HTML.Engine.html_escape()
     end
   end
 
+  if Code.ensure_loaded?(Jason.Encoder) do
```

Comment:
I think we need `ensure_compiled` for third party code, otherwise this can lead to race conditions in Elixir 1.19+ - for JSON itself it should be fine